[REVIEW] Added debug instructions to README. (#664)
rapidsai/cudf: README.md

Done! You are ready to develop for the cuDF OSS project.

## Debugging cuDF

### Building Debug mode from source

Follow the [above instructions](#build-from-source) to build from source and add `-DCMAKE_BUILD_TYPE=Debug` to the `cmake` step. 

For example:
```bash

> $ cmake .. -DCMAKE_INSTALL_PREFIX=/install/path -DCMAKE_BUILD_TYPE=Debug     # configure cmake ... use $CONDA_PREFIX if you're using Anaconda

Reviewer: ```suggestion
$ cmake .. -DCMAKE_INSTALL_PREFIX=/install/path -DCMAKE_BUILD_TYPE=Debug     # configure cmake ... use -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX if you're using Anaconda
```

```

> This will build `libcudf` in Debug mode which enables some `assert` safety checks as well as builds the library with symbols enabled that helps when using a debugger.

Reviewer: ```suggestion
This builds `libcudf` in Debug mode which enables some `assert` safety checks and includes symbols in the library for debugging.
```
